src/components/results/SingleResult.vue

```diff
@@ -1,116 +1,252 @@
-      answer: "answer 3",
-    },
-    {
-      question: "question4",
-      questionScore: 5,
-      answerPoints: 4,
-      category: "category 4",
-      answer: "answer 4",
-    },
-    {
-      question: "question5",
-      questionScore: 5,
-      answerPoints: 0,
-      category: "category 5",
-      answer: "answer 5",
-    },
-    {
-      question: "question6",
-      questionScore: 1,
-      answerPoints: 1,
-      category: "category 6",
-      answer: "answer 6",
-    },
-    {
-      question: "question7",
-      questionScore: 5,
-      answerPoints: 2,
-      category: "category 7",
-      answer: "answer 7",
-    },
-  ],
-  startedAt: 1668092016,
-  endedAt: 1668092016,
-  title: "title 1",
-  id: "1",
-  candidateId: "1",
+defineProps({
+  resultParent: {
+    type: Object as () => ResultParent,
+    default: () => ({
+      position: "",
+      username: "",
+      linkedinUrl: "",
+      feedback: "",
+      avatarUrl: "",
+      id: "",
+    }),
+  },
+  result: {
+    type: Object as () => Result,
+    default: () => ({
+      questionAnswer: [],
+      startedAt: 0,
+      endedAt: 0,
+      id: "",
+      title: "",
+      parent: {
+        position: "",
+        username: "",
+        linkedinUrl: "",
+        feedback: "",
+        avatarUrl: "",
+        id: "",
+      },
+    }),
+  },
+});
+
+const singleResult = ref<Result>({
+  questionAnswer: [],
+  startedAt: 0,
+  endedAt: 0,
+  id: "",
+  title: "",
   parent: {
-    position: "position 1",
-    username: "username 1",
-    linkedinUrl: "linkedinUrl 1",
-    feedback: "feedback 1",
-    avatarUrl:
-      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/602.jpg",
-    id: "1",
+    position: "",
+    username: "",
+    linkedinUrl: "",
+    feedback: "",
+    avatarUrl: "",
+    id: "",
   },
-};
+});
+
+const route = useRoute();
+const { getOneResultForCandidate } = useResultsStore();
+async function getOneResultForCandidateData() {
+  try {
+    const { data } = await getOneResultForCandidate(
+      route.params.candidateId,
+      route.params.resultId,
+    );
+    singleResult.value = { ...data };
+  } catch (e) {
+    console.log(e);
+  }
+}
+getOneResultForCandidateData();
 
 const resultPercentage = computed(() =>
   (
-    (result.questionAnswer.reduce((summ, item) => summ + item.answerPoints, 0) *
-      100) /
-    result.questionAnswer.reduce((summ, item) => summ + item.questionScore, 0)
-  ).toFixed(1),
+    singleResult.value.questionAnswer.reduce(
+      (summ, item) => summ + item.answerPoints,
+      0,
+    ) /
+    singleResult.value.questionAnswer.reduce(
+      (summ, item) => summ + item.point,
+      0,
+    )
+  ).toLocaleString("en-GB", { style: "percent" }),
 );
 
+function answerPointsByCategory(category: string) {
+  const filterAnswers = singleResult.value.questionAnswer.filter(
+    answer => answer.category === category,
+  );
+  const summaryAnswerPoints = filterAnswers.reduce(
```

Comment:
you don't need to declare a vaiable to return the valule. Acually it can be simplified a bit like this
```
return singleResult.value.questionAnswer
    .filter(answer => answer.category === category)
    .reduce((summ, item) => summ + item.answerPoints, 0);
```